File: adjacency_matrix/adjacency_matrix.py
edges = [(1, 4), (2, 7)]
nodes = [1, 4, 2, 7]
number_of_nodes = len(nodes)

adjacency_matrix = [[0] * number_of_nodes for _ in range(number_of_nodes)]

print(adjacency_matrix)

for i in range(number_of_nodes):
    line = nodes[i]
    for j in range(number_of_nodes):
        column = nodes[j]
        edge_draft = (line, column)
        if edge_draft in edges:
            adjacency_matrix[i][j] = 1

print(adjacency_matrix)

File: adjacency_matrix/adjacency_matrix.go
package main

import (
	"fmt"
)

type Edge struct {
	From int
	To   int
}

func main() {
	edges := []Edge{{1, 4}, {2, 7}}
	nodes := []int{1, 4, 2, 7}
	numberOfNodes := len(nodes)

	adjacencyMatrix := make([][]int, numberOfNodes)
	for i := range adjacencyMatrix {
		adjacencyMatrix[i] = make([]int, numberOfNodes)
	}

	fmt.Println("Matriz inicial:", adjacencyMatrix)

	for i := range numberOfNodes {
		line := nodes[i]
		for j := range numberOfNodes {
			column := nodes[j]

			for _, edge := range edges {
				if edge.From == line && edge.To == column {
					adjacencyMatrix[i][j] = 1
				}
			}
		}
	}

	fmt.Println("Matriz final:", adjacencyMatrix)
}
